tracetools/propagate.go

```diff
@@ -22,8 +22,7 @@ func EncodeTraceContext(span opentracing.Span, env map[string]string) error {
 	}
 
 	buf := bytes.NewBuffer([]byte{})
-	enc := gob.NewEncoder(buf)
-	if err := enc.Encode(textmap); err != nil {
+	if err := json.NewEncoder(buf).Encode(textmap); err != nil {
```

Comment:
Switching immediately means there will be a period when people might lose trace linking (but not traces) while they roll out the new agent across their CI fleet. Seemed simpler than trying to plumb through all the env vars and configs, which was what I started with but was having trouble getting working.

Reply:
so i'm not an opentracing expert, but the value of the `BUILDKITE_TRACE_CONTEXT` env var is something that downstream (ie, non-buildkite agent) processes have to decode this to get a trace context, right? it's not just an opaque value?

if this is the case, would the fact that we're switching encodings here means that tracing setups that previously worked would have to change their decoding strategy? if so, this would be a breaking change, and i don't think we'd be able to ship it.

there are other options, i think - we could expose a second env var like `BUILDKITE_TRACE_CONTEXT_BASE64` or something. We could also do the other thing and wire up all the CLI flags and configs and such, though i agree with you that that's not a super great solution; not least of which because it's a pain to implement, but it's also good to keep the number of options on the agent down where we can.

Reply:
I wound up adding an arg and plumbing it through. Turns out the code I had before was mostly fine, I just had typos that was breaking some agent-to-bootstrap plumbing.